src/core/stringdict.cpp

```diff
@@ -13,193 +14,121 @@
  *      - initial implementation and rework communication infrastructure
  *    Martin Jobst
  *      - add string functions accepting a size parameter
+ *    Jörg Walter
+ *      - rework to use std::vector/inplace_vecctor
  *******************************************************************************/
 #include "stringdict.h"
-#include <fortenew.h>
 #ifdef FORTE_ENABLE_GENERATED_SOURCE_CPP
 #include "stringdict_gen.cpp"
 #endif
-#include <stringlist.h>
-#include <string.h>
-#include <stdlib.h>
-#include "devlog.h"
+#include <cstring>
+#include <stringlist.cpp>
 
 DEFINE_SINGLETON(CStringDictionary)
 
-CStringDictionary::CStringDictionary(){
 #ifdef FORTE_STRING_DICT_FIXED_MEMORY
-  mStringBufAddr = scmConstStringBuf;
-  mStringIdBufAddr = scmIdList;
+#include "core/util/inplace_vector.h"
+static forte::core::util::inplace_vector<CStringDictionary::TStringId, FORTE_STRINGDICT_MAXNROFSTRINGS> scmIdList = INDEX_LIST;
```

Comment:
as you moved them out of the class the scm prefix is not valid any more as they are not const and no member.